linode/manager
# fix: [M3-6214] - Reattempt failed HTTP requests in Cypress before hooks

## Description 📝

**What does this PR do?**

- Aims to fix flake that occurs when HTTP requests fail during Cypress before hooks by attempting the clean up process up to 3 times, with 45 second intervals between attempts
- Refactors all of our cleanup functions to use API v4 functions and to run asynchronously, cleans up a lot of unneeded one-off utils and exports
- Refactors polling utils to decouple backoff/retry logic

## How to test 🧪

**Run e2e tests:**
```bash
yarn && yarn build && yarn start:manager:ci
yarn cy:run
```
(Hopefully we'll be able to lean on the PR test results for this)

**To test that before hook reattempts work:**
1. Block API requests by adding the following to the top level of the Cypress config at `packages/manager/cypress.config.ts` (approx `ln 70`)
```typescript
blockHosts: [
  'api.linode.com',
],
```
2. Run `yarn && yarn build && yarn start:manager:ci`
3. Run `yarn cy:debug` and run any test
4. Observe an HTTP 503 error in the before hook, and 2 additional reattempts at approximately 45-second intervals
  * The sidebar should look something like this:
    * <img width="340" alt="Screen Shot 2023-04-06 at 4 41 45 PM" src="https://user-images.githubusercontent.com/97627410/230489914-7aca7d61-fe8f-4e9e-9b82-a935641596e1.png">
